engine/src/globals.cpp

```diff
@@ -832,6 +836,18 @@ void X_clear_globals(void)
 
     MCSocketsInitialize();
     
+    if (MCcommandarguments != NULL)
```

Comment:
This is not correct. The X_clear_globals function is to reset global variables to their initial values - when this is called they should be assumed to be unset. (It is to work around the non-reinit of zero-init data in the android engine on app restart).

All you need to do is: MCcommandarguments / MCcommandname = NULL.